File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        HashSet<Integer> s1=new HashSet<>();
        HashSet<Integer> result=new HashSet<>();

        for(int num:nums1){
            s1.add(num);
        }

        for(int num:nums2){
            if(s1.contains(num)){
                result.add(num);
            }
        }

        int resarr[] = new int[result.size()];
        int index=0;
        for(int setelement:result){
            resarr[index++]=setelement;
        }
        return resarr;
    }
}

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        s1=set(nums1)
        result=set()

        for num in nums2:
            if num in s1:
                result.add(num)
        
        final=list(result)
        return final
        